src/utils/sshUtils.ts

```diff
@@ -6,22 +6,23 @@
 import * as fse from 'fs-extra';
 import * as os from "os";
 import { join } from 'path';
+import { callWithMaskHandling } from 'vscode-azureextensionui';
 import { ext } from '../extensionVariables';
 import { localize } from '../localize';
 import { VirtualMachineTreeItem } from '../tree/VirtualMachineTreeItem';
 import { cpUtils } from "./cpUtils";
 
 export const sshFsPath: string = join(os.homedir(), '.ssh');
 
-export async function getSshKey(vmName: string): Promise<string> {
+export async function getSshKey(vmName: string, passphrase: string = ''): Promise<string> {
     const sshKeyName: string = `azure_${vmName}_rsa`;
     const sshKeyPath: string = join(sshFsPath, sshKeyName);
 
     if (!await fse.pathExists(`${sshKeyPath}.pub`)) {
         // if the SSH key doesn't exist, create it
-        const sshKeygenCmd: string = `ssh-keygen -t rsa -b 2048 -f ${sshKeyPath} -N ""`;
-        ext.outputChannel.appendLog(localize('generatingKey', `Generating public/private rsa key pair with command "${0}".`, sshKeygenCmd));
-        await cpUtils.executeCommand(undefined, undefined, sshKeygenCmd);
+        const sshKeygenCmd: string = `ssh-keygen -t rsa -b 4096 -f ${sshKeyPath} -N "${passphrase}"`;
+        ext.outputChannel.appendLog(localize('generatingKey', 'Generating public/private rsa key pair in directory "{0}"...', sshFsPath));
```

Comment:
When I said I wanted to see `sshKeyPath` in the log, I care more about the name of the key file than the directory. The directory is standard on most machines, but I might have a bunch of different key files in that directory (especially since this extension creates a new one every time), so the file name is the most helpful information